projects/core/src/lib/documentinterruptsource.ts

```diff
@@ -8,7 +11,13 @@ import {
  */
 export class DocumentInterruptSource extends EventTargetInterruptSource {
   constructor(events: string, options?: number | EventTargetInterruptOptions) {
-    super(document.documentElement, events, options);
+    const target =
+      options && (options as EventTargetInterruptOptions).ssr
+        ? null
+        : document.documentElement;
```

Comment:
I'm still getting the below compile error when I import `NgIdleKeepaliveModule.forRoot()` in the app.module of my Angular project, and run the command `npm run build:ssr && npm run serve:ssr`.
```
/Users/bfwg/code/ng2-idle-example/dist/server/main.js:112377
            : document.documentElement;
              ^

ReferenceError: document is not defined
    at new DocumentInterruptSource (/Users/bfwg/code/ng2-idle-example/dist/server/main.js:112377:15)
    at createDefaultInterruptSources (/Users/bfwg/code/ng2-idle-example/dist/server/main.js:112522:9)
    at Module.evxF (/Users/bfwg/code/ng2-idle-example/dist/server/main.js:112527:34)
    at __webpack_require__ (/Users/bfwg/code/ng2-idle-example/dist/server/main.js:20:30)
    at Object.V7fC (/Users/bfwg/code/ng2-idle-example/dist/server/main.js:97581:13)
    at __webpack_require__ (/Users/bfwg/code/ng2-idle-example/dist/server/main.js:20:30)
    at Object.K011 (/Users/bfwg/code/ng2-idle-example/dist/server/main.js:86964:37)
    at __webpack_require__ (/Users/bfwg/code/ng2-idle-example/dist/server/main.js:20:30)
    at Object.0 (/Users/bfwg/code/ng2-idle-example/dist/server/main.js:437:18)
    at __webpack_require__ (/Users/bfwg/code/ng2-idle-example/dist/server/main.js:20:30)
```

`main.js:112377` points to this line.

Reply:
Can you publish this example code somewhere?

Reply:
I made a ssr boilerplate app https://github.com/bfwg/ng2-idle-ssr-example.
Steps to reproduce:
1. clone the ng2-idle-ssr-example app.
2. npm install
3. build `core` and `keepalive` module from thie PR.
4. cope them into the node_modules of the boilerplate app, inside `@ng-idle`
5. run `npm run build:ssr && npm run serve:ssr` command.

Please let me know if you need more information.

Reply:
@bfwg, did you find a solution to your problem? I am having similar errors to yours.